### PR TITLE
update actionHistory entity with announcementEntity

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java [hunk 1]
```diff
@@ -27,332 +27,348 @@
...
 import org.exoplatform.addons.gamification.entities.domain.configuration.AbstractAuditingEntity;
 import org.exoplatform.addons.gamification.entities.domain.configuration.DomainEntity;
+import org.exoplatform.addons.gamification.entities.domain.configuration.RuleEntity;
 import org.exoplatform.commons.api.persistence.ExoEntity;
 
 @ExoEntity
 @Entity(name = "GamificationActionsHistory")
 @Table(name = "GAMIFICATION_ACTIONS_HISTORY")
 @NamedQueries({
-    @NamedQuery(name = "GamificationActionsHistory.findAllActionsHistory", query = "SELECT" +
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java:36]: Could you check that you use the correct formatting configuration ? https://developer.exoplatform.org/#id-ide-settings-to-work-on-exo-platform-projects

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml [hunk 2]
```diff
@@ -296,17 +297,26 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+        <renameColumn oldColumnName="USER_SOCIAL_ID" newColumnName="EARNER_ID" columnDataType="BIGINT"
+                      tableName="GAMIFICATION_ACTIONS_HISTORY"/>
+        <addColumn tableName="GAMIFICATION_ACTIONS_HISTORY">
+            <column name="EARNER_TYPE" type="INT" defaultValueNumeric="0">
+                <constraints nullable="false"/>
+            </column>
+        </addColumn>
+    </changeSet>
+
+    <changeSet author="exo-gamification" id="1.0.0-22">
```
Comment (ahamdi) [inline services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: shouldn't the changeset ID be 1.0.0-21 ?

Comment (GouadriaHanen) [reply services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml:313]: work in progress still need this https://github.com/Meeds-io/gamification/pull/197 to be merged to continue working on this one

File: services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java (hunk 1 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java:36]: done!

File: services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml (hunk 2 above)
Comment (GouadriaHanen) [reply services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml]: done!

**Review by ahamdi — APPROVED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    this.domain = gamificationActionsHistoryEntity.getDomain();
+    this.context = gamificationActionsHistoryEntity.getContext();
+    this.receiver = gamificationActionsHistoryEntity.getReceiver();
+    this.objectId = gamificationActionsHistoryEntity.getObjectId();
+    this.activityId = gamificationActionsHistoryEntity.getActivityId();
+    this.context = gamificationActionsHistoryEntity.getComment();
+    if (gamificationActionsHistoryEntity.getRuleEntity() != null) {
+      this.ruleId = gamificationActionsHistoryEntity.getRuleEntity().getId();
+    }
+  }
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java:57]: use clone please and delete this from DTO
and use lombok

File: services/src/main/java/org/exoplatform/addons/gamification/service/mapper/GamificationActionsHistoryMapper.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import org.exoplatform.addons.gamification.IdentityType;
+import org.exoplatform.addons.gamification.Utils.Utils;
+import org.exoplatform.addons.gamification.entities.domain.effective.GamificationActionsHistory;
+import org.exoplatform.addons.gamification.service.dto.configuration.GamificationActionsHistoryDTO;
+
+public class GamificationActionsHistoryMapper {
+
+  private DomainMapper domainMapper;
+
+  private RuleMapper   ruleMapper;
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/mapper/GamificationActionsHistoryMapper.java:12]: change method public static in DomainMapper & RuleMapper and delete this from the file

File: services/src/main/java/org/exoplatform/addons/gamification/service/mapper/GamificationActionsHistoryMapper.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  public GamificationActionsHistoryMapper(DomainMapper domainMapper, RuleMapper ruleMapper) {
+    this.domainMapper = domainMapper;
+    this.ruleMapper = ruleMapper;
+  }
+
+  public GamificationActionsHistoryDTO fromEntity(GamificationActionsHistory gamificationActionsHistoryEntity) {
+    return new GamificationActionsHistoryDTO(gamificationActionsHistoryEntity);
+  }
+
+  public GamificationActionsHistory toEntity(GamificationActionsHistoryDTO gamificationActionsHistoryDTO) {
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/mapper/GamificationActionsHistoryMapper.java:23]: public static GamificationActionsHistory toEntity(..)

File: services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java [hunk 6]
```diff
@@ -326,33 +299,17 @@ public int hashCode() {
...
-        actionScore +
-        ", receiver=" +
-        receiver +
-        ", objectId=" +
-        objectId +
-        '}';
+    return "Gamification History{" + "date=" + date + ", earner_id=" + earnerId + ", earner_type=" + earnerType
+        + ", global_score=" + globalScore + ", action_title=" + actionTitle + ", domain=" + domain + ", context=" + context
+        + ", action_score=" + actionScore + ", receiver=" + receiver + ", objectId=" + objectId + ", activityId=" + activityId
+        + ", comment=" + comment + '}';
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java:312]: delete this please

File: services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java [hunk 7]
```diff
@@ -27,132 +27,69 @@
...
-})
+        + " FROM GamificationActionsHistory a" + " WHERE a.earnerId = :earnerId"),
+    @NamedQuery(name = "GamificationActionsHistory.getAllPointsByDomain", query = "SELECT g"
+        + " FROM GamificationActionsHistory g" + " WHERE g.domain = :domain "),
+    @NamedQuery(name = "GamificationActionsHistory.getAllPointsWithNullDomain", query = "SELECT g"
+        + " FROM GamificationActionsHistory g" + " WHERE g.domainEntity IS NULL "),
+    @NamedQuery(name = "GamificationActionsHistory.getDomainList", query = "SELECT g.domain"
+        + " FROM GamificationActionsHistory g" + "     GROUP BY  g.domain") })
 public class GamificationActionsHistory extends AbstractAuditingEntity implements Serializable {
   private static final long serialVersionUID = 1L;
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java:94]: use lombok

File: services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
+package org.exoplatform.addons.gamification.service.dto.configuration;
+
+import org.exoplatform.addons.gamification.Utils.Utils;
+import org.exoplatform.addons.gamification.entities.domain.effective.GamificationActionsHistory;
+import java.io.Serializable;
+
+public class GamificationActionsHistoryDTO implements Serializable {
+
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java:8]: implements Cloneable

File: services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  private Long   activityId;
+
+  private String comment;
+
+  public GamificationActionsHistoryDTO() {
+  }
+
+  public GamificationActionsHistoryDTO(GamificationActionsHistory gamificationActionsHistoryEntity) {
+    this.id = gamificationActionsHistoryEntity.getId();
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/dto/configuration/GamificationActionsHistoryDTO.java:41]: change this with clone please

File: services/src/main/java/org/exoplatform/addons/gamification/Utils/Utils.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
+package org.exoplatform.addons.gamification.Utils;
+
```
Comment (Rdidmohamedelhadi) [inline services/src/main/java/org/exoplatform/addons/gamification/Utils/Utils.java:2]: change the name of the package with lower case letters

**Pull request merged**